docs/GraphQL.md

```diff
@@ -0,0 +1,755 @@
+  "type not known/not specified"
+  UNKNOWN
+}
+
+"IsDependency is an attestation to record that a package depends on another."
+type IsDependency {
+  id: ID!
+  "Package that has the dependency"
+  package: Package!
+  "Package for the dependency; MUST BE PackageName, not PackageVersion"
+  dependentPackage: Package!
+  "Version range for the dependency link"
+  versionRange: String!
+  "Type of dependency"
+  dependencyType: DependencyType!
+  "Justification for the attested relationship"
+  justification: String!
+  "Document from which this attestation is generated from"
+  origin: String!
+  "GUAC collector for the document"
+  collector: String!
+}
+```
+
+Note: The version range is currently an opaque string but this will likely
+change in the future.
+
+### "Has a" evidence trees
+
+There are 3 more evidence trees that attest that a software tree has a certain
+property of origin.
+
+The first evidence tree of this type provides the ability to say that a package
+is built from a certain repository:
+
+```gql
+type HasSourceAt {
+  id: ID!
+  "The subject of the attestation: can be a PackageName or a PackageVersion"
+  package: Package!
+  "Source repository from which the package is built"
+  source: Source!
+  "Timestamp since this link between package and source was certified"
+  knownSince: Time!
+  "Justification for the attested relationship"
+  justification: String!
+  "Document from which this attestation is generated from"
+  origin: String!
+  "GUAC collector for the document"
+  collector: String!
+}
+```
+
+We allow for the cases when the source of truth repository is transferred or
+renamed. This also allows for packages in one ecosystem to be built from
+different repositories than packages from another ecosystem (e.g., a similar
+package but with Go sources in one repository and Python sources in another).
+
+The next evidence tree in this section records that an artifact has been built
+by a builder using a certain set of ingredient artifacts, the SLSA attestation:
+
+```gql
+type HasSLSA {
+  id: ID!
+  "The subject of SLSA attestation"
+  subject: Artifact!
+  "The SLSA attestation"
+  slsa: SLSA!
+}
+
+type SLSA {
+  "Materials of the build resulting in subject"
+  builtFrom: [Artifact!]!
+  "Builder performing the build"
+  builtBy: Builder!
+  "Type of the builder"
+  buildType: String!
+  "Individual predicates found in the attestation"
+  slsaPredicate: [SLSAPredicate!]!
+  "Version of the SLSA predicate"
+  slsaVersion: String!
+  "Timestamp (RFC3339Nano format) of build start time"
+  startedOn: Time!
+  "Timestamp (RFC3339Nano format) of build end time"
+  finishedOn: Time!
+  "Document from which this attestation is generated from"
+  origin: String!
+  "GUAC collector for the document"
+  collector: String!
+}
+```
+
+Finally, the last evidence tree defined in GUAC is attaching SBOM information to
+packages and sources:
+
+```gql
+type HasSBOM {
+  id: ID!
+  "SBOM subject"
+  subject: PackageOrSource!
```

Comment:
we will have to make a change here to replace source with artifact

Reply:
@jeffmendoza will do this in subsequent PR